### PR TITLE
Spelling documentation and to do fixes

### PR DESCRIPTION
**Pull request by Tisten**

### CODE REVIEW

File: src/dl_util.cpp [hunk 1]
```diff
@@ -139,8 +139,8 @@ dl_error_t dl_util_load_from_stream( dl_ctx_t dl_ctx,       	dl_typeid_t
 			return DL_ERROR_INTERNAL_ERROR;
 	}
 
-	error = dl_instance_load( dl_ctx, type, load_instance, load_size, load_instance, load_size, 0x0 );
-
```
Comment (wc-duck) [inline src/dl_util.cpp:143]: Not really sure if I like this as this now asume dl_util having knowleage of the internals of dl_load*. The thought about dl util is that it should only use the pure dl-api as is... ie it should maybe mostly be seen as an example.  I'm actually wondering if dl_util should just be replaced with some examples? 
For example I don't use dl_util in my own code as I want to add my own allocators etc...

Comment (Tisten) [reply src/dl_util.cpp:143]: In what way does it now use the internals of "dl_load"? `dl_instance_load_inplace()` is public api, and the old usage of `dl_instance_load()` was "inplace" without actually calling `dl_instance_load_inplace()`.

Comment (Tisten) [reply src/dl_util.cpp:143]: I wouldn't mind replacing utils with examples instead, but the change of memmove -> memcpy feels right

File: src/dl_util.cpp [hunk 2]
```diff
@@ -3,12 +3,34 @@
...
-#include "dl_alloc.h"
-#include "dl_types.h"
 
+#include <stdint.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
-static unsigned char* dl_read_entire_stream( dl_allocator *allocator, FILE* file, size_t* out_size )
+static void patch_alloc_funcs( dl_alloc_func& alloc_func, dl_realloc_func& realloc_func, dl_free_func& free_func )
```
Comment (wc-duck) [inline src/dl_util.cpp:12]: small thing, dl_* prefix.

**Review by wc-duck — APPROVED** [review]

**Pull request merged**